### DISCUSSION TITLE
Validators for Structure

### DISCUSSION THREAD

**Discussion by TobiasPharos** [masqu3rad3/tik_manager4#96] [Ideas]
Would it be possible to add the validator mechanism to the sub-project structure for validating if new sub-projects follow a predefined naming or structure pattern on root-project level?

**Comment by masqu3rad3** [comment]
This can be validated. However it is something different than the DCC publish validations and currently there isn't an exposed mechanism for that.

Can you elaborate on this a bit more with real-world examples? If it is a rule that can be applied to everyone, we can add the rule directly to the core. Otherwise we can think about exposing that feature to the settings level somehow.

**Reply by TobiasPharos** [reply]
I had something in mind that provides an object with the name of the sub-project to be created and its parents, to be able to validate the naming and the position in the structure (I'm using something like a combination of nodes and Lucidity for validation), and expects a boolean if the name and position was verified. So the actual validation process would be in developer's hands.

**not-so-real-world-example:**
User creates a sub-project on level 2 as a child of "Shots", named "1234_5678"

**json:**
{
	"id": 234098604
	"name": "1234_5678"
	"path": "Shots/1234_5678"
}

**Validation Structure (not part of TikManager)**
Shots
    +-- [sequence_name:\d{4}]_[shot_name:\d{3}]

-> The project only accepts a sub-project named "Shots" on level 1. On level 2 the name has to follow the pattern 4 digits, underscore 3 digits. No level 3 allowed.

In this case validation process returns false.

**Result:**
Creation of sub-project fails because the shot name has 4 digits.

**Reply by masqu3rad3** [reply]
Thank you for the explanation, its clear now.

It is not very straightforward to implement a functionality like this as there are no hooks on areas required.

I will think about it and see how it can be implemented.
